File: frontend/gest-vta-web/src/app/services/compania.service.ts
import { HttpClient } from '@angular/common/http';
import { inject, Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { environment } from '../../environments/environment';
import type { Compania } from '../models/api.models';

@Injectable({ providedIn: 'root' })
export class CompaniaService {
  private readonly http = inject(HttpClient);
  private readonly base = `${environment.apiUrl}/api/companias`;

  getAll(): Observable<Compania[]> {
    return this.http.get<Compania[]>(this.base);
  }

  getById(id: number): Observable<Compania> {
    return this.http.get<Compania>(`${this.base}/${id}`);
  }

  create(body: Compania): Observable<Compania> {
    return this.http.post<Compania>(this.base, body);
  }

  update(id: number, body: Compania): Observable<void> {
    return this.http.put<void>(`${this.base}/${id}`, body);
  }

  delete(id: number): Observable<void> {
    return this.http.delete<void>(`${this.base}/${id}`);
  }

  /** Sube logo a FileStorage del servidor; devuelve la ruta web (p. ej. /files/companias/....png) para LogoPath. */
  uploadLogo(file: File): Observable<{ path: string }> {
    const fd = new FormData();
    fd.append('file', file, file.name);
    return this.http.post<{ path: string }>(`${this.base}/logo`, fd);
  }

  /** Elimina físicamente un logo en FileStorage usando su ruta web (/files/companias/...). */
  deleteLogo(path: string): Observable<void> {
    return this.http.delete<void>(`${this.base}/logo`, { params: { path } });
  }
}

File: android/app/src/main/java/com/paucar/webapp/network/CompaniaService.kt
package com.paucar.webapp.network

import com.paucar.webapp.model.Compania
import retrofit2.http.*
import retrofit2.Call
import retrofit2.http.Body
import retrofit2.http.DELETE
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.PUT
import retrofit2.http.Path
import retrofit2.http.Query
import okhttp3.MultipartBody
import okhttp3.RequestBody

interface CompaniaService {
    @GET("api/companias")
    fun getAll(): Call<List<Compania>>

    @GET("api/companias/{id}")
    fun getById(@Path("id") id: Int): Call<Compania>

    @POST("api/companias")
    fun create(@Body compania: Compania): Call<Compania>

    @PUT("api/companias/{id}")
    fun update(@Path("id") id: Int, @Body compania: Compania): Call<Void>

    @DELETE("api/companias/{id}")
    fun delete(@Path("id") id: Int): Call<Void>

    // Upload logo (multipart)
    @Multipart
    @POST("api/companias/logo")
    fun uploadLogo(@Part file: MultipartBody.Part): Call<LogoResponse>

    @DELETE("api/companias/logo")
    fun deleteLogo(@Query("path") path: String): Call<Void>
}

/** Response for logo upload containing the web‑accessible path */
 data class LogoResponse(val path: String)
